### PR TITLE
Codestyle

### PR DESCRIPTION
**Pull request by chenejac**
**[VIVO GitHub issue](https://github.com/vivo-project/VIVO/issues)**: [3814](https://github.com/vivo-project/VIVO/issues/3814)

[Linked Vitro PR](https://github.com/vivo-project/Vitro/pull/372)

* Other Relevant Links (Mailing list discussion, related pull requests, etc.)
https://wiki.lyrasis.org/display/VIVO/Code+Style+Guide

# What does this pull request do?
Adding code style into the VIVO/Vitro project

# What's new?
checkstyle.xml added into the Vitro root
checkstyle-suppression.xml added in the Vitro root (empty at the moment)
A wiki page added - https://wiki.lyrasis.org/display/VIVO/Code+Style+Guide
checkstyle.xml file linked in pom.xml file (at the moment configured to be skipped)

# How should this be tested?
Just for reviewing purposes there is commit with code formatted by using this code style at https://github.com/chenejac/VIVO/commit/863801995c1bcc9ed6685047d17db1c3e88f16e1
Checking style is turned off by definition of suppressions at the level of a Java file:

1. cd VIVO
2. mvn checkstyle:checkstyle
3. No code style violations reported
4. open checkstyle_suppressions.xml and delete line 6 `<suppress files="src[\\/]main[\\/]java[\\/]edu[\\/]cornell[\\/]mannlib[\\/]semservices[\\/]service[\\/]impl[\\/]AgrovocService\.java" checks="."/>`
5. mvn checkstyle:checkstyle
6. Hundreds of code style violations reported

# Interested parties
Tag (@ mention) interested parties or, if unsure, @VIVO-project/vivo-committers

### CODE REVIEW

**Comment by chenejac** [comment]
> Should VIVO have its own checkstyles? Using Vitro from a sibling directory is not desirable.

I suppose there is no any sense in making different style rules for Vitro and VIVO, but we might also copy the same checkstyle.xml from Vitro to VIVO.  @wwelling the practice that Vitro and VIVO projects have to be in sibling directories will be changed in your PRs for improving build process?

**Comment by wwelling** [comment]
@chenejac no that does not change. However, we should try, if possible, to not further the coupling.

**Comment by chenejac** [comment]
> Isn't it possible to build VIVO without Vitro as a sibling directory as long as you do not require any custom changes or specific unpublished Vitro artifact?

Yes, you are right. I have copied checkstyle.xml and checkstyle-suppressions.xml to root of the VIVO project and fixed links in pom.xml file.

**Review by wwelling — DISMISSED** [review]

**Review by wwelling — DISMISSED** [review]
Ignoring comment cleanup as missed last review. ;)

**Review by litvinovg — APPROVED** [review]

File: checkstyle.xml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+<!DOCTYPE module PUBLIC
+        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
+        "http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd">
+<!--
+Vitro CodeStyle Requirements
+
+  1. 4-space indents for Java. NO TABS ALLOWED.
+  2. One true brace style. Braces required on all blocks.
+  3. Do not use wildcard imports (e.g. import java.util.*). Duplicated or unused imports also not allowed.
+  4. Maximum line length is 100 characters (except for long URLs, packages or imports)
```
Comment (litvinovg) [inline checkstyle.xml:11]:
```suggestion
  4. Maximum line length is 120 characters (except for long URLs, packages or imports)
```

File: checkstyle.xml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        <property name="format" value="(?&lt;!\*)\s+$|\*\s\s+$"/>
+        <property name="message" value="Line has trailing whitespace"/>
+        <property name="fileExtensions" value="java, properties, cfg, css, js, xml"/>
+    </module>
+
+    <!-- Allow individual lines of code to be excluded from these rules, if they are annotated
+         with @SuppressWarnings. See also SuppressWarningsHolder below -->
+    <module name="SuppressWarningsFilter" />
+
+    <!-- Maximum line length is 100 characters -->
```
Comment (litvinovg) [inline checkstyle.xml:47]:
```suggestion
    <!-- Maximum line length is 120 characters -->
```

File: checkstyle.xml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    <module name="LineLength">
+        <property name="fileExtensions" value="java"/>
+        <property name="max" value="100"/>
```
Comment (litvinovg) [inline checkstyle.xml:50]:
```suggestion
        <property name="max" value="120"/>
```

**Review by litvinovg — CHANGES_REQUESTED** [review]
Let's increase character limit to 120as we discussed in committers call some time ago.

**Review by litvinovg — DISMISSED** [review]

**Review by litvinovg — APPROVED** [review]

**Pull request merged**